src/gs-updates-page.c

```diff
@@ -1799,9 +1799,11 @@ gs_updates_page_upgrade_cancel_cb (GsUpgradeBanner *upgrade_banner,
 static void
 scheduler_invalidated_cb (GsUpdatesPage *self)
 {
-	g_warning ("The Mogwai Scheduler has just been invalidated when "
-		   "it should still be held!");
-	g_assert_not_reached ();
+	/* The scheduler shouldn’t normally be invalidated, since we Hold() it
+	 * until we’re done with it. However, if the scheduler is stopped by
+	 * systemd (`systemctl stop mogwai-scheduled`) this signal will be
+	 * emitted. */
+	g_clear_pointer (&self->scheduler);
```

Comment:
As [Jenkins has pointed out](https://ci.endlessm-sf.com/job/obs-gnome-software-pr/330/console), it would work better had I actually passed both arguments to `g_clear_pointer()`.